lib/core/Stage.cpp

```diff
@@ -22,6 +22,8 @@
 
 namespace kotekan {
 
+std::map<std::string, pid_t> Stage::thread_list;
```

Comment:
I'd prefer this not be global (in general we avoid using global if at all possible).  I would make this a member variable which just stores the PIDs for each object.   It is possible for a stage to have more than one PID associated with it's unique name (e.g. a stage with more than one thread is allowed).   Then you can just make this a vector, and return a vector.   The `CpuMonitor` class will then need to be passed a reference to the `stages` variable from `KotekanMode`, and iterate that get the list of PID(s) for each stage.